src/PoolManager.sol

```diff
@@ -336,15 +336,22 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
     }
 
     /// @inheritdoc IPoolManager
-    function mint(Currency currency, address to, uint256 amount) external noDelegateCall onlyByLocker {
+    function mint(Currency currency, address to, uint256 amount) external override noDelegateCall onlyByLocker {
         _accountDelta(currency, amount.toInt128());
-        _mint(to, currency, amount);
+        _mint(to, currency.toId(), amount);
     }
 
     /// @inheritdoc IPoolManager
-    function burn(Currency currency, uint256 amount) external noDelegateCall onlyByLocker {
+    function burn(Currency currency, address from, uint256 amount) external override noDelegateCall onlyByLocker {
```

Comment:
now thaat we are 6909, wonder if we should go back to `uint256 tokenId` as the first param here? Bit weird to be able to do `mint(currency, to, amount)` but not be able to do `transfer(currency, to, amount)`

Reply:
Yea I think it is a nice benefit that all funcs on core use Currency

Reply:
Oh hmm, dealing with merge conflicts and it definitely is annoying that `balanceOf` requires uint while mint/burn use Currency.

Reply:
i kind of love a bit of overloading lol, but i know we're keeping v4 core minimal

Reply:
I do think an ordering change would be nice though. So (receiver, id/currency, amount) to mirror 6909?

Reply:
discussed offline: reordered parameters and using Currency for mint/burn + uint256 for other 6909 specific functions